src/com/example/Demo_Subj/Intelligence.java

```diff
@@ -132,12 +135,17 @@ private int getIntRandom(){
     private int manageList(int index, int motivation_highest){
         //es kann nicht das Objekt "beduerfnis" übergeben werden, da sich die eigenschfaten des Objektes ändern müssen. Dies soll dierekt in der Liste geschehen.
 
-        //aktuellen Wert ändern; aktueller Wert = aktueller Wert + Zufallszahl
-        needs_list.elementAt(index).setCurrentValue(needs_list.elementAt(index).getCurrentValue() + getIntRandom());
-
-        //Motivation neu berechnen; Motivation = Priorität * (aktueller Wert - Schwellwert)
-        needs_list.elementAt(index).setMotivation(needs_list.elementAt(index).getPriority() * (needs_list.elementAt(index).getCurrentValue() - needs_list.elementAt(index).getTopLevel()));
+        try {
+            //aktuellen Wert ändern; aktueller Wert = aktueller Wert + Zufallszahl
+            needs_list.elementAt(index).setCurrentValue(needs_list.elementAt(index).getCurrentValue() + getIntRandom());
 
+            //Motivation neu berechnen; Motivation = Priorität * (aktueller Wert - Schwellwert)
+            needs_list.elementAt(index).setMotivation(needs_list.elementAt(index).getPriority() * (needs_list.elementAt(index).getCurrentValue() - needs_list.elementAt(index).getTopLevel()));
+        }
+        catch(Exception e){ //es sollen Exceptions JEDER Art abgefangen werden, da von dne Funktionen nicht expliziert Exceptions geworfen werden
```

Comment:
Till hatte ja gesagt, dass ich den Kern der KI in ein try catch einbetten soll.
Dafür sind entweder Exceptions oder Fehler (Errors) notwendig. Dass keine Exceptions geworfen werden, ist mir schon klar. Deshalb war mir die Sinnhaftigkeit des ganzen auch nicht bewusst.
Errors abfangen habe ich mir gespart, da ich mir einbilde, mal gelesen zu haben, dass sich Java Software im Fehlerfall automatisch beendet. Wieso dann auch abfangen, hab ich mir gedacht...
Mit den Standardwerten annehmen, ist es so eine Sache. In wie weit wäre es denn Sinnvoll Standardwerte zu schreiben, wenn beim Schreibvorgang ein Fehler auftritt?

PS: Entschuldige die späte Antwort. Habe hier Internet nur im Ausnahmefall.

---- Stefan Biereigel schrieb ----

>In src/com/example/Demo_Subj/Intelligence.java:
>
>> > + //Motivation neu berechnen; Motivation = Priorität * (aktueller Wert - Schwellwert) > + needs_list.elementAt(index).setMotivation(needs_list.elementAt(index).getPriority() * (needs_list.elementAt(index).getCurrentValue() - needs_list.elementAt(index).getTopLevel())); > + } > + catch(Exception e){ //es sollen Exceptions JEDER Art abgefangen werden, da von dne Funktionen nicht expliziert Exceptions geworfen werden 
>
>Wenn keine Exceptions geworfen werden, welche sollen dann gefangen werden? Sollte nicht auch irgendwas passieren, wenn ein Fehler passiert ist? Defaults annehmen oder so?
>
>—
>Reply to this email directly or view it on GitHub.￼
>